experiments/formulas/math/avgif.js

```diff
@@ -1,28 +1,31 @@
 // Conditional model column average
 // average conditionally
-// expects input of math__avgif('ModelName','FieldId', 'ConiditionFieldId','ConditionValue') for the corresponding model and field you want to sum conditionally
+// expects input of math__avgif('ModelName','AvgFieldId', 'IfFieldId','IfValue') for the corresponding model and field you want to average conditionally
 // example: I want to find the average Oppty Amount where the Stage is Commit -> math__avgif('Oppty','Amount','StageName','Commit')
 // Matt Davis
 
 skuid.formula.Formula(
 	'avgif',
 	function (modelname, fieldname, iffield, ifvalue) {
 		var model = skuid.$M(modelname),
-			arr = model.data,
+			arr = model.getRows(),
 			initialValue = 0;
 
+		if (arr.length === 0) return 0;
+
 		function filterByField(item) {
 			return item[iffield] === ifvalue;
 		}
 
-		var filtarr = arr.filter(filterByField);
```

Comment:
We need to retain the filtarr array so that we can count it for average in the return statement. Suggest restoring this block.